docs/prepare-osx-devbox.md

```diff
@@ -0,0 +1,54 @@
+# Prepare your development environment using MacOS X
+
+We are open to developers on OSX working on this repository. We are asking macOS users to setup GNU utils on their machines.
+
+We are aiming to limit the amount of shell scripting, etc. in the repository, installing the gnu utils on OSX will minimise the chances of unexpected differences in command line flags, usages, etc., and make it easier for everyone to ensure compatibility down the line.
+
+## Guidances
+
+```bash
+# GNU Utils
+brew install core-utils
+brew search findutils
+
+# Install Jq
+brew install jq
+
+# Install envsubst
+brew install gettext
+brew link --force gettext
+
+# Install
+brew install gpgme
+
+# GNU utils
+# Ref: https://web.archive.org/web/20190704110904/https://www.topbug.net/blog/2013/04/14/install-and-use-gnu-command-line-tools-in-mac-os-x
+# gawk, diffutils, gzip, screen, watch, git, rsync, wdiff
+export PATH="/usr/local/bin:$PATH"
+# coreutils
+export PATH="/usr/local/opt/coreutils/libexec/gnubin:$PATH"
```

Comment:
We probably can get rid of everything apart from `coreutils` and `findutils` here (or do we need something else?)

I had rest of them in my `PATH` because I installed more GNU utils on my machine for Kubernetes upstream development. If we don't need them for the project and if we don't install them - probably it doesn't make sense to mention them here. What do you think?